pubnub/models/subscription.py

```diff
@@ -37,13 +38,13 @@ def is_matching_listener(self, message):
         def wildcard_match(name, subscription):
             return subscription.endswith('.*') and name.startswith(subscription.strip('*'))
 
-        if self.subscription_type == PNSubscriptionType.CHANNEL:
+        if self._type == PNSubscriptionType.CHANNEL:
             return any(
                 [message.channel == subscription or wildcard_match(message.channel, subscription)
-                    for subscription in self.subscription_names]
+                    for subscription in self.name]
             )
-        elif self.subscription_type == PNSubscriptionType.CHANNEL_GROUP:
-            return message.subscription in self.subscription_names
+        elif self._type == PNSubscriptionType.CHANNEL_GROUP:
+            return message.subscription in self.name
         else:
```

Comment:
is this the branch for subscription sets?
my Python is not great, but will this really work?
does a subscriptionset have self.channels and self.channel_groups?
does adding names of channels and groups make sense? one should be checked in `message.channel`, the other in `message.subscription` according to the code above, but here you're checking both in `message.channel`?

I might be missing how this works, so correct me if I'm wrong here

Reply:
I don't see definitions of these two fields either. 
Perhaps the same is for the `_type` field - where do you get it from?

Reply:
So this checker operates on all subscriptions regardless of where they come from single channel/group or subscription sets. When you receive a message from channel in channel group the value of group is in `.subscription` property so when you have channel_group it should only check the value of this one field